dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/archaius/impl/CachedCompositeProperty.java

```diff
@@ -116,21 +114,41 @@ public T get() {
             }
 
             if (cache.compareAndSet(currentValue, newValue, cacheVersion, latestVersion)) {
-                return firstNonNull(newValue, defaultValue);
+                compositeValue = newValue;
+                fromCache = false;
             }
         }
-        return firstNonNull(cache.getReference(), defaultValue);
+        if (fromCache) {
+            compositeValue = cache.getReference();
+        }
+
+        return compositeValue;
     }
 
     /**
-     * Select a non null preferably from two values
+     * Fetch the latest version of the property. If not up to date then resolve to the latest
+     * value, inline.
+     *
+     * TODO: Make resolving property value an offline task
```

Comment:
We should not cargo copy this TODO over into our method as we do not want to DO that.